src/lib.rs

```diff
@@ -0,0 +1,528 @@
+#[cfg(not(target_os = "windows"))]
+use jemallocator::Jemalloc;
+use router::Router;
+use tracing_subscriber::{layer::SubscriberExt, util::SubscriberInitExt, Layer};
+#[cfg(not(target_os = "windows"))]
+#[global_allocator]
+static GLOBAL: Jemalloc = Jemalloc;
+
+use crate::shared::utils::AbortOnDrop;
+use config::Configuration;
+use key_utils::Secp256k1PublicKey;
+use lazy_static::lazy_static;
+use proxy_state::{PoolState, ProxyState, TpState, TranslatorState};
+use self_update::{backends, cargo_crate_version, update::UpdateStatus, TempDir};
+use std::sync::OnceLock;
+use std::{net::SocketAddr, time::Duration};
+use tokio::sync::mpsc::channel;
+use tracing::{debug, error, info, warn};
+
+mod api;
+mod config;
+mod ingress;
+pub mod jd_client;
+mod minin_pool_connection;
+mod monitor;
+mod proxy_state;
+mod router;
+mod share_accounter;
+mod shared;
+mod translator;
+
+const TRANSLATOR_BUFFER_SIZE: usize = 32;
+const MIN_EXTRANONCE_SIZE: u16 = 6;
+const MIN_EXTRANONCE2_SIZE: u16 = 5;
+const UPSTREAM_EXTRANONCE1_SIZE: usize = 20;
+const DEFAULT_SV1_HASHPOWER: f32 = 100_000_000_000_000.0;
+const CHANNEL_DIFF_UPDTATE_INTERVAL: u32 = 10;
+const MAX_LEN_DOWN_MSG: u32 = 10000;
+const MAIN_AUTH_PUB_KEY: &str = "9c44K6QVizyPWb9xfeqhckFRosxWwB3EfytGa4CfTdD526qb2QV";
+const TEST_AUTH_PUB_KEY: &str = "9auqWEzQDVyd2oe1JVGFLMLHZtCo2FFqZwtKA5gd9xbuEu7PH72";
+const DEFAULT_LISTEN_ADDRESS: &str = "0.0.0.0:32767";
+const REPO_OWNER: &str = "demand-open-source";
+const REPO_NAME: &str = "demand-cli";
+const BIN_NAME: &str = "demand-cli";
+const STAGING_URL: &str = "https://staging-user-dashboard-server.dmnd.work";
+const LOCAL_URL: &str = "http://localhost:8787";
+const TESTNET3_URL: &str = "https://testnet3-user-dashboard-server.dmnd.work";
+const PRODUCTION_URL: &str = "https://production-user-dashboard-server.dmnd.work";
+
+lazy_static! {
+    static ref SV1_DOWN_LISTEN_ADDR: String =
+        Configuration::downstream_listening_addr().unwrap_or(DEFAULT_LISTEN_ADDRESS.to_string());
+    static ref TP_ADDRESS: roles_logic_sv2::utils::Mutex<Option<String>> =
+        roles_logic_sv2::utils::Mutex::new(Configuration::tp_address());
+    static ref POOL_ADDRESS: roles_logic_sv2::utils::Mutex<Option<SocketAddr>> =
+        roles_logic_sv2::utils::Mutex::new(None); // Connected pool address
+    static ref EXPECTED_SV1_HASHPOWER: f32 = Configuration::downstream_hashrate();
+    static ref API_SERVER_PORT: String = Configuration::api_server_port();
```

Comment:
I wonder if we need this as global variable? can u please check? It should be used in single place, and if so - we should not have it as global variable.